fish-rust/src/common.rs

```diff
@@ -1717,16 +1717,6 @@ fn get_executable_path(argv0: &str) -> PathBuf {
     std::env::current_exe().unwrap_or_else(|_| PathBuf::from_str(argv0).unwrap())
 }
 
-/// Like [`std::mem::replace()`] but provides a reference to the old value in a callback to obtain
-/// the replacement value. Useful to avoid errors about multiple references (`&mut T` for `old` then
-/// `&T` again in the `new` expression).
-pub fn replace_with<T, F: FnOnce(&T) -> T>(old: &mut T, with: F) -> T {
-    let new = with(old);
-    std::mem::replace(old, new)
-}
```

Comment:
Yes, this was no longer used once @krobelus introduced `scoped_push()`; it was an ergonomic workaround used when there was only `ScopeGuard` and should have been dropped before. I mentioned it elsewhere but the `allow(unused)` feature flag prevented the compiler from reporting this.